tests/foreman/cli/test_import.py

```diff
@@ -102,3 +110,124 @@ def test_reimport_orgs_default(self):
                 'csv-file': files['users']
             }).stdout, [u'Summary', u'  No action taken.', u'']
         )
+
+    @skip_if_bug_open('bugzilla', 1160847)
+    def test_BZ1160847(self):
+        """@test: Check whether all supported Sat5 macros are being
+        properly converted to the Puppet facts.
+        According to RH Transition Guide (Chapter 3.7.8, Table 3.1)
+
+        @feature: Import config-file --csv-file --generate-only
+
+        @assert: Generated .erb file contains correctly formated puppet facts
+
+        @BZ: 1160847
+
+        """
+        # prepare data (craft csv)
+        data = [
+            {
+                u'name': u'hostname',
+                u'macro': u'{| rhn.system.hostname |}',
+                u'fact': u'<%= @fqdn %>',
+            },
+            {
+                u'name': u'sys_ip_address',
+                u'macro': u'{| rhn.system.ip_address |}',
+                u'fact': u'<%= @ipaddress %>',
+            },
+            {
+                u'name': u'ip_address',
+                u'macro': u'{| rhn.system.net_interface'
+                          u'.ip_address(eth0) |}',
+                u'fact': u'<%= @ipaddress_eth0 %>',
+            },
+            {
+                u'name': u'netmask',
+                u'macro': u'{| rhn.system.net_interface'
+                          u'.netmask(eth0) |}',
+                u'fact': u'<%= @netmask_eth0 %>',
+            },
+            {
+                u'name': u'mac_address',
+                u'macro': u'{| rhn.system.net_interface.'
+                          u'hardware_address(eth0) |}',
+                u'fact': u'<%= @macaddress_eth0 %>',
+            },
+        ]
+        csv_contents = u'\n'.join(
+            u'{0}={1}'.format(i['name'], i['macro']) for i in data
+        )
+        csv_header = [
+            u'org_id', u'channel_id', u'channel', u'channel_type', u'path',
+            u'file_type', u'file_id', u'revision', u'is_binary', u'contents',
+            u'delim_start', u'delim_end', u'username', u'groupname',
+            u'filemode', u'symbolic_link', u'selinux_ctx'
+        ]
+        csv_row = [
+            u'1', u'3', u'config-1', u'normal',
+            u'/etc/sysconfig/rhn/systemid', u'file', u'8', u'1', u'N',
+            csv_contents + u'\n',
+            u'{|', u'|}', u'root', u'root', u'600', u'', u''
+        ]
+
+        file_name = tempfile.mkstemp()[1]
+        with open(file_name, 'wb') as csv_file:
+            csv_writer = csv.writer(
+                csv_file,
+                delimiter=',',
```

Comment:
No need to define delimiter, quotechar and quoting. The default dialect already does that, see below:

```python
In [6]: w = csv.writer(open('test.csv', 'w'))

In [8]: w.dialect.delimiter
Out[8]: ','

In [9]: w.dialect.quotechar
Out[9]: '"'

In [11]: w.dialect.quoting == csv.QUOTE_MINIMAL
Out[11]: True
```